cpp/src/copying/shift.cu

```diff
@@ -60,7 +60,7 @@ struct shift_functor {
     auto& scalar     = static_cast<ScalarType const&>(fill_value);
 
     auto device_input  = column_device_view::create(input);
-    auto output        = allocate_like(input, mask_allocation_policy::NEVER);
+    auto output        = allocate_like(input, mask_allocation_policy::NEVER, mr);
```

Comment:
Don't we need to pass stream as well ?

Reply:
As I already mentioned in the pull request, [`allocate_like`](https://github.com/rapidsai/cudf/blob/branch-0.14/cpp/include/cudf/copying.hpp#L168-L171) doesn't take a stream. Until that's fixed, I can't really pass the stream here.

Reply:
https://github.com/rapidsai/cudf/blob/b12f24d25815145a573a9d70f8c6140a8ab9d2cb/cpp/include/cudf/detail/copy.hpp#L100

Reply:
Sure, I can use that instead. I was just worried that the non-detail one is not a thin wrapper; remember this was the case in my previous pull request and we ended up not passing the stream. And I think we shouldn't be calling the detail method directly in general, as there is no guarantee that the API-exposed one is always a simple wrapper like this.

Reply:
I just pushed a change to use the detail method you suggested. Feel free to resolve the conversation if that looks good to you.